services:
  postgres:
    image: postgres:17.5
    container_name: postgres
    ports:
      - "5432:5432"
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}" ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - bot-network
    restart: unless-stopped

  bot-app:
    image: ghcr.io/vddmit/telegram-ai-chat-provider:${IMAGE_TAG}
    container_name: telegram-ai-bot
    depends_on:
      postgres:
        condition: service_healthy
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
      BOT_TOKEN: ${BOT_TOKEN}
      BOT_NAME: ${BOT_NAME}
    networks:
      - bot-network
    restart: unless-stopped

  nginx-certbot:
    image: jonasal/nginx-certbot:latest
    container_name: nginx-certbot-proxy
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
    environment:
      - CERTBOT_EMAIL=danvoropaeff@yandex.ru
      - STAGING=0
    volumes:
      - nginx_secrets:/etc/letsencrypt
      - ./user_conf.d:/etc/nginx/user_conf.d:ro
    depends_on:
      - bot-app
    networks:
      - bot-network

volumes:
  postgres_data:

  nginx_secrets:

networks:
  bot-network:
    driver: bridge